### ISSUE TITLE
DeleteRecords returns empty response and does not delete

### ISSUE THREAD

**Issue by rmb938** [twmb/franz-go#235]
Library Version Info
```
github.com/twmb/franz-go v1.9.0
github.com/twmb/franz-go/pkg/kadm v1.3.0
github.com/twmb/franz-go/pkg/kmsg v1.2.0
```

Kafka Version 3.1.2

I'm not sure if I found a bug but I am not able to get DeleteRecords to work. 

I can delete records via the kafka cli using the following perfectly fine:

```
./bin/kafka-delete-records.sh --bootstrap-server $HOSTNAME.$K8S_SERVICE_CONNECTION_STRING:$CONTAINER_PORT --command-config /opt/kafka/config/helper.properties --offset-json-file partitions.json
Executing records delete operation
Records delete operation completed:
partition: __CruiseControlMetrics-0	low_watermark: 4987758


# cat partitions.json
{"partitions":
                                         [{"topic": "__CruiseControlMetrics", "partition": 0,
                                         "offset": -1}],
                                        "version":1
                                       }
```

But when I try and use DeleteRecords, it returns no error and the response is empty.

See logs and code bellow:
```
1.6667264675349133e+09 INFO controller.kafkatopicpurge Getting topic details for purge {"reconciler group": "pephccdp.optum.com", "reconciler kind": "KafkaTopicPurge", "name":
"purge-2", "namespace": "example"}
1.6667264677768514e+09 INFO controller.kafkatopicpurge Adding partition to purge {"reconciler group": "pephccdp.optum.com", "reconciler kind": "KafkaTopicPurge", "name": "purge-2",
"namespace": "example", "partition": 0}
1.6667264677768958e+09 INFO controller.kafkatopicpurge Purging {"reconciler group": "pephccdp.optum.com", "reconciler kind": "KafkaTopicPurge", "name": "purge-2", "namespace":
"example", "offsets": {"__CruiseControlMetrics":{"0":{"Topic":"__CruiseControlMetrics","Partition":0,"At":-1,"LeaderEpoch":0,"Metadata":""}}}}
1.6667264680108159e+09 INFO controller.kafkatopicpurge Purge response {"reconciler group": "pephccdp.optum.com", "reconciler kind": "KafkaTopicPurge", "name": "purge-2", "namespace":
"example", "resp": {}, "err": null}
```

Sample Code, this is inside a custom Kubernetes controller so feel free to ignore those bits:
```golang
			rLog.Info("Getting topic details for purge")
			topicDetails, err := kadmClient.ListTopics(ctx, kafkaTopicPurge.Spec.TopicName)
			if err != nil {
				return ctrl.Result{}, err
			}

			topicDetailResp := topicDetails[kafkaTopicPurge.Spec.TopicName]
			if topicDetailResp.Err != nil {
				// TODO: make this an event as well
				//   we probably want to figure out how to put something into the status as well
				//   because customers may not notice and the events could be gone by the time we look
				rLog.Error(topicDetailResp.Err, "error listing topic", "topic", kafkaTopicPurge.Spec.TopicName)
				kafkaTopicPurge.Status.Phase = pephccdpv1alpha1.TopicPurgeFailed
				err = r.Status().Update(ctx, kafkaTopicPurge)
				if err != nil {
					return ctrl.Result{}, err
				}
				return ctrl.Result{}, nil
			}

			offsets := make(kadm.Offsets)
			offsets[kafkaTopicPurge.Spec.TopicName] = make(map[int32]kadm.Offset)

			errored := false
			for _, detail := range topicDetailResp.Partitions {
				if detail.Err != nil {
					// TODO: make this an event as well
					//   we probably want to figure out how to put something into the status as well
					//   because customers may not notice and the events could be gone by the time we look
					rLog.Error(detail.Err, "error listing topic partition", "topic", kafkaTopicPurge.Spec.TopicName, "partition", detail.Partition)
					errored = true
					continue
				}

				rLog.Info("Adding partition to purge", "partition", detail.Partition)
				offsets[kafkaTopicPurge.Spec.TopicName][detail.Partition] = kadm.Offset{
					Topic:     kafkaTopicPurge.Spec.TopicName,
					Partition: detail.Partition,
					At:        -1,
				}
			}

			if errored {
				kafkaTopicPurge.Status.Phase = pephccdpv1alpha1.TopicPurgeFailed
				err = r.Status().Update(ctx, kafkaTopicPurge)
				if err != nil {
					return ctrl.Result{}, err
				}
				return ctrl.Result{}, nil
			}

			rLog.Info("Purging", "offsets", offsets)
			response, err := kadmClient.DeleteRecords(ctx, offsets)
			if err != nil {
				return ctrl.Result{}, err
			}

			errored = false

			rLog.Info("Purge response", "resp", response, "err", err)
```

**Comment by twmb** [comment]
If I'm reading my code correctly, the bug looks to be a dumb one:
```go
355         req := kmsg.NewPtrDeleteRecordsRequest()
356         req.TimeoutMillis = cl.timeoutMillis
357         for t, ps := range os {
358                 rt := kmsg.NewDeleteRecordsRequestTopic()
359                 rt.Topic = t
360                 for p, o := range ps {
361                         rp := kmsg.NewDeleteRecordsRequestTopicPartition()
362                         rp.Partition = p
363                         rp.Offset = o.At
364                 }
365         }
```
rp is not appended to rt.Partitions, and rt is not appended to req.Topics.

Fix:
```
diff --git a/pkg/kadm/topics.go b/pkg/kadm/topics.go
index c93b46e..8f425bd 100644
--- a/pkg/kadm/topics.go
+++ b/pkg/kadm/topics.go
@@ -361,7 +361,9 @@ func (cl *Client) DeleteRecords(ctx context.Context, os Offsets) (DeleteRecordsR
                        rp := kmsg.NewDeleteRecordsRequestTopicPartition()
                        rp.Partition = p
                        rp.Offset = o.At
+                       rt.Partitions = append(rp.Partitions, rp)
                }
+               req.Topics = append(req.Topics, rt)
        }

        shards := cl.cl.RequestSharded(ctx, req)
```

**Comment by twmb** [comment]
(I blame writing most of this package in a one day blitz at a cafe)

**Comment by rmb938** [comment]
Ah, I even read through the code and didn't notice that.

**Comment by twmb** [comment]
released in v1.3.1, forgot to give credit in the tag, but thank you

**Comment by rmb938** [comment]
No problem! Thanks for the quick replies and review :)
